docs/content/concepts/grpc-overview.mdx

```diff
@@ -0,0 +1,378 @@
+
+In the Sui gRPC API, FieldMasks are used in requests like `GetTransaction` to control which parts of the transaction (e.g., `effects`, `events`) are included in the response. Field paths must match the structure of the response message. This selective querying is especially useful for building efficient applications and tools.
+
+### Encoding
+
+In the Sui gRPC API, identifiers with standard human-readable formats are represented as `string`s in the proto schema:
+
+- `Address` and `ObjectId`: Represented as 64 hex characters with a leading `0x`.
+- `Digest`s: Represented as [Base58](https://learnmeabitcoin.com/technical/keys/base58/).
+- `TypeTag` and `StructTag`: Represented in their canonical string format (e.g. `0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::sui::SUI>`)
+
+## Sample Sui gRPC clients
+
+### Access using grpcurl
+
+Simplest way to experiment with gRPC is by using [grpcurl](https://github.com/fullstorydev/grpcurl).
+
+:::note
+
+Your results may be different from the examples shared below, depending on the breadth and maturity of the gRPC APIs available on Sui full nodes.
+
+:::
+
+**List available gRPC services**
+
+```shell
+grpcurl <full node URL:port> list
+```
+
+where the port on Sui Foundation managed full nodes is `443`. It should return something like:
+
+```shell
+grpc.health.v1.Health
+grpc.reflection.v1.ServerReflection
+sui.rpc.v2alpha.LiveDataService
+sui.rpc.v2alpha.SubscriptionService
+sui.rpc.v2beta.LedgerService
+sui.rpc.v2beta.TransactionExecutionService
+```
+
+**List available APIs in the LedgerService**
+
+```shell
+grpcurl <full node URL:port> list sui.rpc.v2beta.LedgerService
+```
+
+which should return something like:
+
+```shell
+sui.rpc.v2beta.LedgerService.BatchGetObjects
+sui.rpc.v2beta.LedgerService.BatchGetTransactions
+sui.rpc.v2beta.LedgerService.GetCheckpoint
+sui.rpc.v2beta.LedgerService.GetEpoch
+sui.rpc.v2beta.LedgerService.GetObject
+sui.rpc.v2beta.LedgerService.GetServiceInfo
+sui.rpc.v2beta.LedgerService.GetTransaction
+```
+
+**Get the `events` and `effects` details of a particular transaction**
+
+```shell
+grpcurl -d '{ "digest": "3ByWphQ5sAVojiTrTrGXGM5FmCVzpzYmhsjbhYESJtxp" }' <full node URL:port> sui.rpc.v2beta.LedgerService/GetTransaction
+```
+
+**Get the transactions in a particular checkpoint**
+
+```shell
+grpcurl -d '{ "sequence_number": "180529334", "read_mask": { "paths": ["transactions"]} }' <full node URL:port> sui.rpc.v2beta.LedgerService/GetCheckpoint
+```
+
+### Access using typescript
+
+This is an example to build a typescript client for Sui gRPC API. Feel free to use another set of tools or modules that you’re comfortable with.
+
+**Install dependencies**
+
+```shell
```

Comment:
Put each of these in their own codeblock. The site has a copy code button, which fails if the user tries to copy and run as is when there are multiple commands. And use `sh` instead. And previously mentioned $ sign.
```
'''sh
$ npm init -y
'''
'''sh
$ npm install...
'''
```